### PR TITLE
resolve_issues-feets_to_metric-hazard_materials_included-fail_with_tr…

### PR DESCRIPTION
**Pull request by Ypsilonx**
Hello,

I was recently asked by a mapping friend that there are some ambiguities in your plugin, which is absolutely great.
Since he doesn't deal with python and wanted to make it work he asked me if I would take a look at it.

I looked at it and made a change request - hopefully I did this all right - this is my first attempt at such a change on github.

Thanks for the feedback.

### CODE REVIEW

File: valhalla/gui/common_gui.py [hunk 1]
```diff
@@ -73,21 +73,21 @@ def get_costing_options(costing_group, profile):
...
+    costing_widgets = costing_group.findChildren((QComboBox, QSpinBox, QDoubleSpinBox, QCheckBox))
     costing_widgets = [o for o in costing_widgets if profile in o.objectName()]
 
     costing_options = dict()
     for widget in costing_widgets:
         param_name = widget.objectName().replace(profile + '_', '')
         if isinstance(widget, (QDoubleSpinBox, QSpinBox)):
             if widget.objectName() == 'truck_length' or widget.objectName() == 'truck_width':
-                costing_options[param_name] = round(widget.value() / 3.28084, 2)
+                costing_options[param_name] = widget.value()
```
Comment (radekvermirovsky) [inline valhalla/gui/common_gui.py:84]: IMO there is no need to keep IF for truck length and width at all so lets just keep only IF for spinboxes

` if isinstance(widget, (QDoubleSpinBox, QSpinBox)):
costing_options[param_name] = widget.value()`

File: valhalla/proc/request_builder.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def get_costing_options(costing_options, profile, mode):
         for cost in costing_options:
             if cost[1]:
                 if cost[0] in [TRUCK_COSTING.WIDTH, TRUCK_COSTING.LENGTH]:  # Temp bcs of Valhalla issue
-                    params[profile][cost[0]] = round(cost[1] / 3.28084, 2)
+                    params[profile][cost[0]] = cost[1]
```
Comment (radekvermirovsky) [inline valhalla/proc/request_builder.py:106]: again no need to keep IF for width and length so remove it completly:
```python
if cost[1]:
   params[profile][cost[0]] = cost[1]
```

File: valhalla/gui/common_gui.py (hunk 1 above)
Comment (Ypsilonx) [reply valhalla/gui/common_gui.py]: Sorry, I left the wrong version of the file during the test.

It should be fixed now.

Thank you very much for the feedback.

File: valhalla/proc/request_builder.py (hunk 2 above)
Comment (Ypsilonx) [reply valhalla/proc/request_builder.py]: Same problem.

It should be fixed now.

Thank you very much for the feedback.

File: valhalla/gui/common_gui.py [hunk 3]
```diff
@@ -73,21 +73,18 @@ def get_costing_options(costing_group, profile):
...
-            if widget.objectName() == 'truck_length' or widget.objectName() == 'truck_width':
-                costing_options[param_name] = round(widget.value() / 3.28084, 2)
-            else:
-                costing_options[param_name] = widget.value()
+            costing_options[param_name] = widget.value()
         if isinstance(widget, QComboBox):
             if widget.currentText():
                 costing_options[param_name] = widget.currentText()
         if isinstance(widget, QCheckBox):
-            costing_group[param_name] = widget.isChecked()
```
Comment (nilsnolde) [inline valhalla/gui/common_gui.py:91]: haha what was `costing_group` even referring to?

File: valhalla/gui/common_gui.py [hunk 4]
```diff
@@ -73,21 +73,18 @@ def get_costing_options(costing_group, profile):
...
     costing_widgets = [o for o in costing_widgets if profile in o.objectName()]
 
     costing_options = dict()
     for widget in costing_widgets:
         param_name = widget.objectName().replace(profile + '_', '')
         if isinstance(widget, (QDoubleSpinBox, QSpinBox)):
-            if widget.objectName() == 'truck_length' or widget.objectName() == 'truck_width':
-                costing_options[param_name] = round(widget.value() / 3.28084, 2)
-            else:
-                costing_options[param_name] = widget.value()
```
Comment (nilsnolde) [inline valhalla/gui/common_gui.py:86]: I remember this. I think the problem is somewhat related to https://github.com/valhalla/valhalla/issues/4473. Probably we were testing HERE in the US back then and it just was a weird conversion.

File: valhalla/proc/costing_params.py [hunk 5]
```diff
@@ -239,15 +239,14 @@ def set_costing_options(self, proc_algo, parameters, context):
                 )
             )
 
-        # Boolean hazmat
         setattr(
             self,
             TRUCK_COSTING.HAZMAT,
-            proc_algo.parameterAsBool(
+            bool(proc_algo.parameterAsBool(
```
Comment (nilsnolde) [inline valhalla/proc/costing_params.py:245]: it's fine to cast again, but `parameterAsBool` should already return a `bool`: https://qgis.org/pyqgis/3.0/core/Processing/QgsProcessingAlgorithm.html#qgis.core.QgsProcessingAlgorithm.parameterAsBool. if not, there's smth wrong with pyqgis.

**Review by nilsnolde — APPROVED** [review]
Didn't look at this code in ages, but your changes look fine and safe enough. Thanks!

File: valhalla/gui/common_gui.py (hunk 4 above)
Comment (Ypsilonx) [reply valhalla/gui/common_gui.py:86]: Is it possible to work out a conversion to feet - if someone would like not to?

**Pull request merged**